app/models/course/condition/assessment.rb

```diff
@@ -89,4 +94,13 @@ def required_assessments_for(conditional)
       ON ids.assessment_id = course_assessments.id
     SQL
   end
+
+  # Trigger for resolving the conditional for a course user
+  Course::Assessment::Submission.after_save do |submission|
```

Comment:
this should follow the rest of the models in terms of their ordering. See contributing.md, put this as the last callback for the condition::assessment model.